Dz8 additional

angryQ opened the pull request.

angryQ committed:
Initial

angryQ committed:
Добавила кнопку старт/стоп для проигрывания картинок и перезапуск с нужной картинки

Не разобралась откуда после нажатия на стоп + старт берется пауза и как от нее избавиться.
Также после нажатия на стоп успевает показаться следующая картинка и на ней происходит остановка

angryQ committed:
Набросок UIStepper и UISlider

Разобраться как сделать чтобы размытие элемента не выходило за пределы UIImage и почему при нажатии на кнопки степпера в некоторых случаях пропускается одна картинка

angryQ committed:
Добавила UIProgressView

Закинула все кнопки в стек вью. Отличается индексация картинок в степпере и при нажатии на старт/стоп, не могу найти место где происходит рассинхронизация индекса

angryQ committed:
Починила синхронизацию индексов картинок и добавила оставшиеся задания с выбором даты и UISwitch

ios-review commented on DZ8-additional/DZ8-additional/DatesViewController.swift:
```diff
@@ -0,0 +1,41 @@
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .full
+        dateFormatter.dateFormat = "d MMMM, YYYY HH:mm"
+        dateFormatter.timeZone = TimeZone(abbreviation: "GMT" + timeZoneTextField.text!)
```
можно ввести значение +100, что не является временной зоной

ios-review commented on DZ8-additional/DZ8-additional/Dates.storyboard:
```diff
@@ -0,0 +1,83 @@
+                                    <action selector="datePickerChanged:" destination="Afk-LP-Bhu" eventType="valueChanged" id="wCs-cd-9zs"/>
+                                </connections>
+                            </datePicker>
+                            <textField opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="left" contentVerticalAlignment="center" borderStyle="roundedRect" placeholder="f.e. -5" textAlignment="natural" minimumFontSize="17" translatesAutoresizingMaskIntoConstraints="NO" id="PLX-9M-2Og">
```
e. g. - сокращение для for example
для placeholder

ios-review commented on DZ8-additional/DZ8-additional/GalleryViewController.swift:
```diff
@@ -0,0 +1,128 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        galleryImageView.image = images[0]
```
`images.first`

ios-review commented on DZ8-additional/DZ8-additional/GalleryViewController.swift:
```diff
@@ -0,0 +1,128 @@
+    @IBOutlet weak var stepper: UIStepper!
+    @IBOutlet weak var progressView: UIProgressView!
+    
+    let animator = UIViewPropertyAnimator(duration: 0.5, curve: .linear)
```
☺️

ios-review commented on DZ8-additional/DZ8-additional/GalleryViewController.swift:
```diff
@@ -0,0 +1,128 @@
+    //События выполняемые во время запущенной анимации
+    private func startAnimationFromImageWithIndex(_ index: Int) {
+        let subArray = Array(self.images[index..<self.images.count])
+        self.galleryImageView.animationImages = subArray
+        self.galleryImageView.animationDuration = Double(subArray.count) * duration
+        self.galleryImageView.startAnimating()
```
Если использовать таймер, то можно просто менять значение image у UIImageView.
Так должно быть проще.

ios-review commented on DZ8-additional/DZ8-additional/GalleryViewController.swift:
```diff
@@ -0,0 +1,128 @@
+    }
+    //Обновляем состояние прогресс вью
+    private func updateProgressView() {
+        self.progressView.progress =  0.1 * Float(currentPicIndex + 1)
```
Не очень интуитивно:

<img width="522" alt="Screenshot 2020-11-12 at 14 44 00" src="https://user-images.githubusercontent.com/62285448/98911106-a98bd380-24f6-11eb-93ad-d38f82bfb14e.png">

ios-review commented on DZ8-additional/DZ8-additional/ViewController.swift:
```diff
@@ -0,0 +1,20 @@
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+    }
+
+
+}
+
```
Пустой класс.

ios-review reviewed (changes requested):
Комментарии к кометам - топ👍
Это хорошая практика писать дескриптивные комментарии к коду!
Молодец🔥

По заданию 👌

Чуть-чуть можно упростить + добавить вариацию на тайм зону.

angryQ committed:
Добавила правки по рекомендациям Дмитрия

Добавила обработку неверных данных введенных в поле с таймзоной и небольшие правки по галерее

ios-review commented on DZ8-additional/DZ8-additional/DatesViewController.swift:
```diff
@@ -16,26 +16,42 @@ class DatesViewController: UIViewController {
+    
+    private func checkTextInTextField(text: String) -> String {
+        guard let textInt = Int(text) else {return "\(formatDate(date: datePicker.date))"}
+        if textInt >= -12 && textInt <= 14 {
```
Почему 14?😏

ios-review commented on DZ8-additional/DZ8-additional/DatesViewController.swift:
```diff
@@ -16,26 +16,42 @@ class DatesViewController: UIViewController {
+        guard let _ = Int(checkTextInTextField(text: timeZoneTextField.text!)) else {
+            resultLabel.text = checkTextInTextField(text: timeZoneTextField.text!)
+            return }
+        resultLabel.text = formatDate(date: datePicker.date, timezone: TimeZone(abbreviation: "GMT" + timeZoneTextField.text!))
```
При вводе значения больше 0, значение не меняется(

ios-review replied:
Кроме значений больше 14

ios-review commented on DZ8-additional/DZ8-additional/GalleryViewController.swift:
```diff
@@ -27,7 +27,7 @@ class GalleryViewController: UIViewController {
+        galleryImageView.image = images.first
         galleryImageView.contentMode = .scaleAspectFill
         startButton.setTitle("Стоп", for: .selected)
         slider.value = 0
```
При открытии приложение падает(

На скрине подсказка. Было непросто найти проблему)

<img width="1407" alt="Screenshot 2020-11-18 at 18 13 19" src="https://user-images.githubusercontent.com/62285448/99523541-c8dca200-29c9-11eb-8e6b-b3ae3ced16bc.png">

ios-review reviewed (changes requested):
Появилось пара багов.

angryQ committed:
Правки по рекомендациям Дмитрия

ios-review commented on DZ8-additional/DZ8-additional/DatesViewController.swift:
```diff
@@ -25,23 +25,22 @@ class DatesViewController: UIViewController {
+        case -12...(-1): resultLabel.text = formatDate(date: datePicker.date, timezone: TimeZone(abbreviation: "GMT" + "\(textInt)"))
+        default:
+            return resultLabel.text = "Введите число от -12 до +12 (включительно)"
         }
```
14 было верно)
Хотел, чтобы вы пруф привели)

🔥

ios-review commented on DZ8-additional/DZ8-additional/Gallery.storyboard:
```diff
@@ -77,7 +77,6 @@
                         <outlet property="slider" destination="Ury-hI-HV0" id="TZD-OC-SM5"/>
                         <outlet property="startButton" destination="k71-Z2-BiG" id="Ya5-hb-Upx"/>
                         <outlet property="stepper" destination="4He-lQ-g4C" id="Rdn-Fq-Ql7"/>
-                        <outlet property="view" destination="MV6-wq-fZM" id="f6k-My-aA0"/>
```
Свойство view не может быть заменено, так как это имя занято корневой вью для контроллера.

ios-review reviewed (approved):
Супер гуд🚀

ПР можно мержить.
